File: app/src/main/java/com/wade/libs/FileTools.java
package com.wade.libs;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.File;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;

/**
  FileTools fileTools = new FileTools(Environment.getExternalStorageDirectory()+"/Status.txt");
  fileTools.append("PWR_ON,"+ Calendar.getInstance(timeZone).getTime().toString()+"\n");
 */

public class FileTools {
    private String fileName;
    public FileTools(String fn) {
        fileName = fn;
    }
    public FileTools() {
        fileName = "";
    }
    public void setFileName(String fn) {
        fileName = fn;
    }
    public String load() {
        File file = new File(fileName);
        if (!file.exists()) return "";
        try {
            FileReader f = new FileReader(file);
            BufferedReader bufferedReader = new BufferedReader(f);
            String text = "";
            String buf="";
            while ((buf = bufferedReader.readLine()) != null) {
                text += buf+"\n";
            }
            bufferedReader.close();
            f.close();
            return text;
        } catch (IOException e) {
        }
        return "";
    }

    public void delete() {
        File file = new File(fileName);
        file.delete();
    }

    private void write(String buf, boolean append) {
        try {
            File file = new File(fileName);
            FileWriter f  = new FileWriter(file, append);
            BufferedWriter bufferedWriter = new BufferedWriter(f);

            bufferedWriter.write(buf);
            bufferedWriter.flush();
            bufferedWriter.close();
            f.close();
        } catch (IOException e) {
        }
    }

    public void save(String buf) {
        write(buf, false);
    }
    public void append(String buf) {
        write(buf, true);
    }
    public void empty() {
        write("", false);
    }
}

File: app/src/main/java/com/wade/libs/FileTools.kt
package com.wade.libs

import android.util.Log
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.File
import java.io.FileReader
import java.io.FileWriter
import java.io.IOException

/**
 * FileTools fileTools = new FileTools(Environment.getExternalStorageDirectory()+"/Status.txt");
 * fileTools.append("PWR_ON,"+ Calendar.getInstance(timeZone).getTime().toString()+"\n");
 */
class FileTools {
    private var fileName: String

    constructor(fn: String) {
        fileName = fn
    }

    constructor() {
        fileName = ""
    }

    fun setFileName(fn: String) {
        fileName = fn
    }

    fun load(): String {
        val file = File(fileName)
        if (!file.exists()) return ""
        try {
            val f = FileReader(file)
            val bufferedReader = BufferedReader(f)
            var text = ""
            var buf = ""
            while (bufferedReader.readLine().also { buf = it } != null) {
                text += """
                    $buf
                    
                    """.trimIndent()
            }
            bufferedReader.close()
            f.close()
            return text
        } catch (e: IOException) {
            Log.d("MyLog", "'" + fileName + "' 讀檔錯誤: " + e.message)
        }
        return ""
    }

    fun delete() {
        val file = File(fileName)
        file.delete()
    }

    private fun write(buf: String, append: Boolean) {
        try {
            val file = File(fileName)
            val f = FileWriter(file, append)
            val bufferedWriter = BufferedWriter(f)
            bufferedWriter.write(buf)
            bufferedWriter.flush()
            bufferedWriter.close()
            f.close()
        } catch (e: IOException) {
            Log.d("MyLog", "'" + fileName + "' 寫檔錯誤: " + e.message)
        }
    }

    fun save(buf: String) {
        write(buf, false)
    }

    fun append(buf: String) {
        write(buf, true)
        Log.d("MyLog", "append: '" + buf.trim { it <= ' ' } + "'")
    }

    fun empty() {
        write("", false)
    }
}